src/Raven.Server/Documents/Indexes/Static/IndexCompiler.cs

```diff
@@ -468,6 +464,24 @@ static Assembly LoadAssembly(string path)
 
                 return Assembly.LoadFile(path);
             }
+
+            static void RegisterPackage(NuGetFetcher.NuGetPackage package, bool userDefined, HashSet<MetadataReference> references)
+            {
+                if (package == null)
+                    return;
+
+                foreach (string library in package.Libraries)
+                {
+                    var assembly = LoadAssembly(library);
+                    references.Add(RegisterAssembly(assembly));
+                }
+
+                if (userDefined)
+                    NuGetNativeLibraryResolver.RegisterPath(package.NativePath);
```

Comment:
Need to make a decision if we want to register only native DLLs from packages that user defined or also from dependencies. I'm a bit afraid to 'let it loose' and register everything.